flink-tests/src/test/java/org/apache/flink/test/checkpointing/SavepointFormatITCase.java

```diff
@@ -78,138 +82,151 @@
     LoggerAuditingExtension loggerAuditingExtension =
             new LoggerAuditingExtension(SavepointFormatITCase.class, Level.INFO);
 
-    private static Stream<Arguments> parameters() {
-        return Stream.of(
-                Arguments.of(
-                        SavepointFormatType.CANONICAL,
-                        HEAP,
-                        (Consumer<KeyedStateHandle>)
-                                keyedState ->
-                                        assertThat(
-                                                keyedState,
-                                                instanceOf(SavepointKeyedStateHandle.class))),
-                Arguments.of(
-                        SavepointFormatType.NATIVE,
-                        HEAP,
-                        (Consumer<KeyedStateHandle>)
-                                keyedState ->
-                                        assertThat(
-                                                keyedState,
-                                                instanceOf(KeyGroupsStateHandle.class))),
-                Arguments.of(
-                        SavepointFormatType.CANONICAL,
-                        ROCKSDB_FULL_SNAPSHOTS,
-                        (Consumer<KeyedStateHandle>)
-                                keyedState ->
-                                        assertThat(
-                                                keyedState,
-                                                instanceOf(SavepointKeyedStateHandle.class))),
-                Arguments.of(
-                        SavepointFormatType.NATIVE,
-                        ROCKSDB_FULL_SNAPSHOTS,
-                        (Consumer<KeyedStateHandle>)
-                                keyedState ->
-                                        assertThat(
-                                                keyedState,
-                                                instanceOf(KeyGroupsStateHandle.class))),
-                Arguments.of(
-                        SavepointFormatType.CANONICAL,
-                        ROCKSDB_INCREMENTAL_SNAPSHOTS,
-                        (Consumer<KeyedStateHandle>)
-                                keyedState ->
-                                        assertThat(
-                                                keyedState,
-                                                instanceOf(SavepointKeyedStateHandle.class))),
-                Arguments.of(
-                        SavepointFormatType.NATIVE,
-                        ROCKSDB_INCREMENTAL_SNAPSHOTS,
-                        (Consumer<KeyedStateHandle>)
-                                keyedState ->
-                                        assertThat(
-                                                keyedState,
-                                                instanceOf(
-                                                        IncrementalRemoteKeyedStateHandle.class))));
+    private static List<Arguments> parameters() {
```

Comment:
Only expressing my opinion, I'll leave it up to your judgment. I see the benefit of generating all combinations through "loops", but on the downside I find it harder to 1) see what is tested, 2) add new test cases or which case/branch applies to which backend combination.

Reply:
I agree that the assertions are now disconnected from the inputs.
But looking at it from a different perspective, this allows to understand what the end result should be via `validateState`; for example, `CANONICAL` savepoints always result in  `SavepointKeyedStateHandle`.

So I'd leave it with the loops.